multiqc/modules/freyja/freyja.py

```diff
@@ -0,0 +1,145 @@
+            if s_name in self.freyja_data:
+                log.debug("Duplicate sample name found! Overwriting: {}".format(s_name))
+            self.freyja_data[s_name] = d
+            self.add_data_source(f, s_name)
+
+    def general_stats_cols(self):
+        """Add a single column displaying the most abundant lineage to the General Statistics table"""
+        top_lineage_dict = {}
+        top_lineages = set()
+        for s_name, sub_dict in self.freyja_data.items():
+            top_lineage = max(sub_dict, key=sub_dict.get)
+            top_lineage_value = sub_dict[top_lineage]
+            top_lineage_dict[s_name] = {
+                "Top_lineage_freyja": top_lineage,
+                "Top_lineage_freyja_percentage": top_lineage_value,
+            }
+            top_lineages.add(top_lineage)
+
+        headers = OrderedDict()
+        scale = mqc_colour.mqc_colour_scale("Pastel2", 0, len(top_lineages))
+        headers["Top_lineage_freyja"] = {
+            "title": "Top lineage",
+            "description": "The most abundant lineage in the sample",
+            "bgcols": {val: scale.get_colour(i) for i, val in enumerate(top_lineages)},
+        }
+        headers["Top_lineage_freyja_percentage"] = {
+            "title": "Top lineage %",
+            "description": "The percentage of the most abundant lineage in the sample",
+            "max": 100,
+            "min": 0,
+            "scale": "Blues",
+            "modify": lambda x: x * 100,
+            "suffix": "%",
+        }
+
+        self.general_stats_addcols(top_lineage_dict, headers)
+
+    def add_freyja_section(self):
+        cats = list()
+        pconfig = {
+            "id": "Freyja_plot",
+            "title": "Freyja: Top lineages",
+            "ylab": "relative abundance",
+            "yCeiling": 1,
+            "cpswitch": False,
+            "cpswitch_c_active": False,
+        }
+
+        rank_cats = OrderedDict()
+
+        rank_cats["other"] = {"name": "Other", "color": "#cccccc"}
+        cats.append(rank_cats)
+
+        self.add_section(
+            name="Freyja Summary",
+            anchor="freyja-summary",
+            description="""
+                Relative lineage abundances from mixed SARS-CoV-2 samples. Hover over the column headers for descriptions and click _Help_ for more in-depth documentation. 
+                """,
+            helptext="""
+                The graph denotes a sum of all lineage abundances in a particular WHO designation , otherwise they are grouped into "Other".
+                Lineages abundances are calculated as the number of reads that are assigned to a particular lineage. 
+                Lineages and their corresponding abundances are summarized by constellation. 
+
+                > **Note**: Lineage designation is based on the used WHO nomenclature, which could vary over time. 
+                """,
+            plot=bargraph.plot(self.freyja_data, None, pconfig),
```

Comment:
It would be good to make sure the bars have the same sorting order, e.g. on bar 2 Omicron goes at the tail, and on the bar 3 it is in the head.

<img width="1033" alt="Screenshot 2023-08-30 at 16 23 09" src="https://github.com/ewels/MultiQC/assets/1575412/662c28dd-d174-45e5-8100-95ac72b4ec42">

Reply:
Sorting order was simple (pushed a change), however, making sure the barchart colors are the same as the general stats column colors is not obvious: for the column, more pastel colors look better, whereas on the barchart, the bright default highcharts colors look the best.